metaci/conftest.py

```diff
@@ -1,51 +1,107 @@
 import factory
 import factory.fuzzy
 
+import numbers
+import random
+
 from metaci.plan.models import Plan, PlanRepository
 from metaci.testresults.models import TestResult, TestMethod, TestClass
-from metaci.build.models import BuildFlow, Build
+from metaci.build.models import BuildFlow, Build, BUILD_STATUSES, BUILD_FLOW_STATUSES
 from metaci.repository.models import Branch, Repository
 
 from metaci.users.models import User
 
+BUILD_STATUS_NAMES = (name for (name, label) in BUILD_STATUSES)
+BUILD_FLOW_STATUS_NAMES = (name for (name, label) in BUILD_FLOW_STATUSES)
+
+
+def do_logs():
+    import logging
+
+    logger = logging.getLogger("factory")
+    logger.addHandler(logging.StreamHandler())
+    logger.setLevel(logging.DEBUG)
+
+
+def fake_name(prefix=None):
+    return factory.LazyAttribute(
+        lambda a: (getattr(a, "_name_prefix", None) or prefix or "")
+        + factory.Faker("word").generate({})
+    )
+
+
+class RelatedFactoryList(factory.RelatedFactory):
+    """Calls a factory 'size' times once the object has been generated.
+        Simplified from
+        https://github.com/FactoryBoy/factory_boy/blob/2481d411cde311bb7edd51b6dff3b345c2c14bc6/factory/declarations.py#L675
+        While awaiting release
+    """
+
+    def __init__(self, factory, factory_related_name="", size=2, **defaults):
+        self.size = size
+        super(RelatedFactoryList, self).__init__(
+            factory, factory_related_name, **defaults
+        )
+
+    def call(self, instance, step, context):
+        return [
+            super(RelatedFactoryList, self).call(instance, step, context)
+            for i in range(self.size if isinstance(self.size, int) else self.size())
+        ]
+
 
 class PlanFactory(factory.django.DjangoModelFactory):
     class Meta:
         model = Plan
 
+    _name_prefix = "Plan"
+    name = fake_name()
+
 
 class RepositoryFactory(factory.django.DjangoModelFactory):
     class Meta:
         model = Repository
+        exclude = ("_name_prefix",)
```

Comment:
Is there a reason not to just pass the prefix in as an argument to `fake_name`? That would avoid setting the `_name_prefix` attribute, so would avoid the need to exclude it here. But I suppose I'm missing something or else you would have done that...

Reply:
Maybe its overly theoretical, but it allows the constructor of the RepositoryFactory to override the name, which is in-line with the general style of FactoryBoy where you can push in stuff from outside. I'll experiment with it to ensure that it really works and really adds value in the data generated (e.g. labelling Public repos and Private repos to ensure that the private ones are hidden in the public view). If it doesn't add value or doesn't work right then I'll take it out.

Reply:
Ah okay, that makes sense. Though if it's for external use it should probably not start with an underscore.

Reply:
https://github.com/SFDO-Tooling/MetaCI/pull/408/commits/1d3326acfe3421222ca126d5aa753580690dad6e